File: src/main/java/nox/sys/OS.java
/**
 * Copyright (c): 2017 Oleg Sklyar and contributors. License: MIT
 */
package nox.sys;

import org.gradle.internal.os.OperatingSystem;


public enum OS {
	linux("linux"),
	win32("win32"),
	macosx("macosx");

	private final String value;

	OS(String value) {
		this.value = value;
	}

	public static OS current() {
		OperatingSystem os = OperatingSystem.current();
		return os.isWindows() ? win32 : (os.isMacOsX() ? macosx : linux);
	}

	public static boolean current(OS check) {
		return current() == check;
	}

	@Override
	public String toString() {
		return value;
	}
}

File: src/main/kotlin/nox/sys/OS.kt
/**
 * Copyright (c): 2017 Oleg Sklyar and contributors. License: MIT
 */
package nox.sys

import org.gradle.internal.os.OperatingSystem


enum class OS private constructor(private val value: String) {
	linux("linux"),
	win32("win32"),
	macosx("macosx");

	override fun toString(): String {
		return value
	}

	companion object {

		fun current(): OS {
			val os = OperatingSystem.current()
			return if (os.isWindows) win32 else if (os.isMacOsX) macosx else linux
		}

		fun current(check: OS): Boolean {
			return current() == check
		}
	}
}
